apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Data.Common;
+using System.Threading.Tasks;
+using Xunit;
+// ReSharper disable RedundantArgumentDefaultValue
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    public class WriteTests : IClassFixture<TestDatabaseFixture>
+    {
+        private readonly TestDatabaseFixture _testFixture;
+        private string _lastKey;
+
+        public WriteTests(TestDatabaseFixture testFixture)
+        {
+            _testFixture = testFixture;
+        }
+
+        private static String UniqueString()
+        {
+            return Guid.NewGuid().ToString();
+        }
+
+        private async Task<int> Insert(SpannerParameterCollection values)
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                values.Add("K", _lastKey = UniqueString(), SpannerDbType.String);
+                var cmd = connection.CreateInsertCommand("T", values);
+                
+                return await cmd.ExecuteNonQueryAsync();
+            }
+        }
+
+        private async Task<DbDataReader> GetLastRowAsync()
+        {
+            using (var connection = await _testFixture.GetTestDatabaseConnectionAsync())
+            {
+                var cmd = connection.CreateSelectCommand("SELECT * FROM T WHERE K=@k",
+                    new SpannerParameterCollection{ {"K", _lastKey, SpannerDbType.String }});
+                var reader = await cmd.ExecuteReaderAsync();
+                await reader.ReadAsync();
+                return reader;
```

Comment:
Yes this is actually ok.  There are actually options in ado.net that allow you to specify that the reader closes the connection when it finishes.  I just allow for it either way and don't force you to send in the option.